src/Access/AnalyzeAccessAccessChecker.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\analyze\Access;
+
+use Drupal\analyze\AnalyzeTrait;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Checks access against enabled plugins for Analyze report routes.
+ */
+final class AnalyzeAccessAccessChecker implements AccessInterface {
+
+  use AnalyzeTrait;
+
+  /**
+   * Access callback for Analyze routes.
+   *
+   * @param \Symfony\Component\Routing\Route $route
+   *   The current route.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The current logged in user.
+   * @param string $entity_type
+   *   The entity type the route relates to.
+   * @param string|null $plugin
+   *   The current plugin being views (optional).
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function access(Route $route, AccountInterface $account, string $entity_type, string|null $plugin = NULL): AccessResult {
+    if ($entity = $this->getEntity($entity_type)) {
+      if ($account->hasPermission('view analyze reports')) {
+        $return = AccessResult::forbidden('Entity not enabled for Analyze reporting.');
+
+        if ($config = $this->analyzeConfig()) {
+          if ($settings = $config->get('status')) {
+            if (!empty($settings[$entity_type])) {
+              if (!empty($settings[$entity_type][$entity->bundle()])) {
+
+                // If we are looking at a specific plugin, check it as been
+                // enabled before granting access.
+                if ($plugin) {
+                  if (!empty($settings[$entity_type][$entity->bundle()][$plugin])) {
+                    if ($plugins = $this->getPlugins([$plugin])) {
+
+                      $analyze = reset($plugins);
+
+                      // If the plugin modifies the full report URL, we should
+                      // deny access to our default route.
+                      if (!$analyze->fullReportUrlOverridden($entity)) {
```

Comment:
I think we can simply check whether `fullReportUrl` is _not_ `NULL`. If it's overridden with a non-null, we won't be using the route that this access check is attached to anyway, so I think we can simply assume if we have gotten to this position, we can allow. That was we can drop `fullReportUrlOverridden` entirely.

Reply:
If we just check whether it's null, in cases where people have implemented a custom URL and returned it, we'll have our default URL kicking around doing nothing - this was done so we can disable the default route and its action links if all it's going to do is show an empty page.